src/main/java/cz/it4i/parallel/ui/HPCSettingsScreenWindow.java

```diff
@@ -0,0 +1,92 @@
+
+package cz.it4i.parallel.ui;
+
+
+
+import cz.it4i.parallel.runners.HPCSettings;
+import javafx.scene.Scene;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import javafx.stage.Window;
+
+public class HPCSettingsScreenWindow 
+{
+
+	private HPCSettingsScreenController controller;
+
+	private HPCSettings settings;
+
+	private Window owner;
+
+	public HPCSettings showDialog(final HPCSettings oldSettings) {
+		// Get the old settings:
+
+		if (oldSettings != null) {
+			this.settings = oldSettings;
+		}
+		this.controller = new HPCSettingsScreenController();
+		setInitialTextFieldText();
+		// Request new settings:
+		this.openWindow();
+
+		// If the user did not provide new settings:
+		if (controller.getSettings() == null) {
+			// Return old settings.
+			return this.settings;
+		}
+		// Return the new settings.
+		return controller.getSettings();
+	}
+
+	public void setOwner(Window aOwner) {
+		this.owner = aOwner;
+	}
+
+	private void openWindow() {
+		final Scene fileSelectionScene = new Scene(controller);
+		final Stage parentStage = new Stage();
+		parentStage.initModality(Modality.APPLICATION_MODAL);
+		parentStage.setResizable(false);
+		parentStage.setTitle("HPC Settings");
+		parentStage.setScene(fileSelectionScene);
+		parentStage.initOwner(owner);
+		// Set the text fields to the old settings:
+
+		parentStage.showAndWait();
+	}
+
+	private void setInitialTextFieldText() {
```

Comment:
Consider its moving it to ...Controller

Reply:
I move this method and the equivalent method of the "LocalSettingsScreenWindow" in their controllers.